### PR TITLE
Adding livestream messagelist

### PR DESCRIPTION
**Pull request by vishalnarkhede**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: src/components/MessageList/MessageList.js [hunk 1]
```diff
@@ -32,6 +32,7 @@ class MessageList extends PureComponent {
     this.state = {
```
Comment (vishalnarkhede) [inline src/components/MessageList/MessageList.js:32]: Adding this file only to make it easier to see diff. This file will be reverted after the review

File: src/components/MessageList/LivestreamMessageList.js [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+      this.setState({
+        slicedMessages: this.props.messages.slice(
+          -(this.state.slicedMessages.length + this.props.messageLimit),
+        ),
+      });
+      return;
+    }
+
+    // Otherwise fetch from server.
+    if (this.props.messageLimit) {
```
Comment (tatianainama) [inline src/components/MessageList/LivestreamMessageList.js:260]: Not really familiar with the code, but is this condition necessary? You end up calling `props.loadMore` with an `undefined` value for `messageLimit` anyways

Comment (mahboubii) [reply src/components/MessageList/LivestreamMessageList.js:260]: passing `undefined` down will replace the `loadMore(limit = 100)` default value, good call though, we can simplify this a bit

Comment (vishalnarkhede) [reply src/components/MessageList/LivestreamMessageList.js]: Makes sense ... have fixed it by adding default value to messageLimit prop and removing this if condition

File: src/components/MessageList/LivestreamMessageList.js [hunk 3]
```diff
@@ -0,0 +1,502 @@
...
+
+    if (scrollTop !== undefined) containerEl.scrollTop = scrollTop; // eslint-disable-line no-param-reassign
+  };
+
+  goToNewMessages = () => {
+    this.scrollToBottom();
+    this.setState({ newMessagesNotification: false });
+  };
+
+  userScrolledUp = () => this.scrollOffset > 310;
```
Comment (tatianainama) [inline src/components/MessageList/LivestreamMessageList.js]: Maybe a nitpick, but what does this `310` means?

File: src/components/MessageList/LivestreamMessageList.js [hunk 4]
```diff
@@ -0,0 +1,502 @@
...
+  /**
+   * target - target to scroll to (DOM element, scrollTop Number, 'top', or 'bottom'
+   * containerEl - DOM element for the container with scrollbars
+   * source: https://stackoverflow.com/a/48429314
+   */
+  scrollToTarget = (target, containerEl) => {
+    // Moved up here for readability:
+    const isElement = target && target.nodeType === 1;
+    const isNumber =
+      Object.prototype.toString.call(target) === '[object Number]';
```
Comment (tatianainama) [inline src/components/MessageList/LivestreamMessageList.js]: Using `toString()` to check types is unreliable (the `toStringTag` can be changed to anything). How about using `typeof` instead?

**Review by jaapbakker88 — APPROVED** [review]

**Pull request closed without merging**